detectable/src/main/java/com/synopsys/integration/detectable/detectables/poetry/parser/ToolPoetrySectionParser.java

```diff
@@ -55,29 +55,33 @@ public Set<String> parseRootPackages(File pyprojectToml, PoetryOptions options)
         }
 
         for (String key : parseResult.dottedKeySet(true)) {
-            if (!parseResult.isTable(key)) {
-                continue;
-            }
+            processKeyForRootPackages(parseResult, options, result, key);
+        }
+
+        result.remove(PYTHON_COMPONENT_NAME);
+        return result;
+    }
+
+    private void processKeyForRootPackages(TomlParseResult parseResult, PoetryOptions options, Set<String> result, String key) {
+        if (!parseResult.isTable(key)) {
+            return;
+        }
 
-            TomlTable table = parseResult.getTable(key);
+        TomlTable table = parseResult.getTable(key);
 
-            if (key.equals(MAIN_DEPENDENCY_GROUP_KEY)) {
+        if (key.equals(MAIN_DEPENDENCY_GROUP_KEY)) {
+            addAllTableKeysToSet(result, table);
+        } else if (key.equals(LEGACY_DEV_DEPENDENCY_GROUP_KEY)) { // in Poetry 1.0 to 1.2 this was the way of specifying dev dependencies
+            if (!options.getExcludedGroups().contains(DEFAULT_DEV_GROUP_NAME)) {
```

Comment:
Both can be combined into the same `else if` with `&&`

Reply:
Thanks, Nirav. I prefer to keep it this way so that the comment on line 74 refers to one specific thing unambiguously.